### PR TITLE
Alerting API specification

### PR DESCRIPTION
**Pull request by anmolbabu**
tendrl-bug-id: https://github.com/Tendrl/specifications/issues/123
Signed-off-by: anmolbabu <anmolbudugutta@gmail.com>

### CODE REVIEW

**Comment by anmolbabu** [comment]
@Tendrl/tendrl-core @Tendrl/tendrl_api @anivargi Please review

File: specs/alerting-api.adoc [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+
+The rest apis for:
+
+* Listing and filtering alerts
+* List different supported alert notification medium
+* Provide a way to know the necessary and optional configuration attributes
+  of different alert notification medium.
+* List all supported types of alerts.
+
+need to be exposed by tendrl-api.
```
Comment (shtripat) [inline specs/alerting-api.adoc:16]: Do we need this line ?

File: specs/alerting-api.adoc [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+
+==== Tendrl API impact:
+
+The following apis need to be exposed by the tendrl-api module:
+
+* This adds the following apis:
+  ** Api to get list of currently supported means of notification.
+
+----
+GET /alerting/notification_medium/supported
```
Comment (shtripat) [inline specs/alerting-api.adoc:131]: Do we need api to be `/alerting/notification_medium/supported` or `/alerting/notification_medium` is enough ?

File: specs/alerting-api.adoc [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+{
+  notif_medium: [email, snmp]
+}
+----
+
+  ** Api to get list of alerts with various filtering options such as based on
+     time, acked/not acked, alert type, severity, resource and significance.
+
+----
+GET /alerting/alerts?severity=CRITICAL
```
Comment (shtripat) [inline specs/alerting-api.adoc:145]: May be we should list all possible filters which could be used

**Pull request merged**